src/canvasObject.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var CanvasObject = function(config) {
+  this.fullyOpened = config.fullyOpened || false;
+  this.visible = config.visible || true;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.x = config.x || 0;
+  this.y = config.y || 0;
+  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
+
+  this.id = config.canvas['@id'];
+  this.height = config.canvas.height;
+  this.width = config.canvas.width;
+  this.images = config.canvas.images;
+  this.label = config.canvas.label;
+  this.tileSourceUrl = config.canvas.images[0].resource.service['@id'] + '/info.json';
+};
+
+CanvasObject.prototype = {
+  openTileSource: function(viewer) {
+    // We've already loaded this tilesource instead of the thumbnail
+    if(this.fullyOpened) {
+      return;
+    }
+    var self = this;
+    var bounds = self.mainImageObj.getBounds();
+    viewer.viewport.fitBounds(bounds);
```

Comment:
This `fitBounds` is just here because currently `openTileSource` is only called when the user clicks on the image...in the future we'll probably call it in other contexts, so the `fitBounds` should probably come out of here. 

Also, CanvasObject probably shouldn't be in charge of moving the viewport around at all...it should just keep track of its own self.